### ISSUE TITLE
ERROR in Cannot read property 'provide' of null 

### ISSUE THREAD

**Issue by Codermar** [angular/angular-cli#5208]
> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
Mac OSX Sierra

### Versions.
@angular/cli: 1.0.0-rc.1
node: 6.10.0
os: darwin x64
@angular/common: 2.4.9
@angular/compiler: 2.4.9
@angular/compiler-cli: 2.4.9
@angular/core: 2.4.9
@angular/forms: 2.4.9
@angular/http: 2.4.9
@angular/platform-browser: 2.4.9
@angular/platform-browser-dynamic: 2.4.9
@angular/router: 3.4.9
@angular/upgrade: 2.4.9
@angular/cli: 1.0.0-rc.1

### Repro steps.
Run npm start or ng build

### The log given by the failure.
Hash: a27f48afc01f81613fbf
Time: 18192ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 228 kB {5} [initial] [rendered]
chunk    {1} main.bundle.js, main.bundle.js.map (main) 480 kB {4} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 285 kB {5} [initial] [rendered]
chunk    {3} scripts.bundle.js, scripts.bundle.js.map (scripts) 465 kB {5} [initial] [rendered]
chunk    {4} vendor.bundle.js, vendor.bundle.js.map (vendor) 3.69 MB [initial] [rendered]
chunk    {5} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in Cannot read property 'provide' of null
webpack: Failed to compile.

### Mention any other details that might be useful.

* If I edit any file that triggers a recomplie, everything works.
* ng build produces the same error

I am running  "node-sass": "^3.13.1".

When I run ng buld --prod I get:
Hash: f09bb9d74690b8b7a885
Time: 13625ms
chunk    {0} polyfills.6bb8b84172f155217606.bundle.js (polyfills) 228 kB {5} [initial] [rendered]
chunk    {1} scripts.0e2daf9bda10b6d206e1.bundle.js (scripts) 465 kB {5} [initial] [rendered]
chunk    {2} styles.9258649ea7b6ee560358.bundle.css (styles) 281 bytes {5} [initial] [rendered]
chunk    {3} main.daff98269318dace91c3.bundle.js (main) 668 bytes {4} [initial] [rendered]
chunk    {4} vendor.094b768043954986be41.bundle.js (vendor) 834 kB [initial] [rendered]
chunk    {5} inline.f79b29e42b419838a9c7.bundle.js (inline) 0 bytes [entry] [rendered]

ERROR in Cannot read property 'text' of undefined

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/usr/local/apple/bullwinkle-jive/apple-xquad-bullwinkle/bullwinkle-ui/src'
 @ ./src/main.ts 4:0-74
 @ multi ./src/main.ts

My main.ts is pretty much out of the box from the CLI

**Comment by filipesilva** [comment]
I can't reproduce this on a new project. If you can get me some reproduction steps where I can see this issue happen myself, I can try to figure out what the problem is.

**Comment by Codermar** [comment]
Neither can I on a new project. This happens on a project that's been around for a bit. I am investigating further. Thanks.

**Comment by filipesilva** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular CLI version.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.

**Comment by peteichuk** [comment]
Wel @bastienJS I did get rid of the error. I tried building it with the -aot option so ng build -aot
This gave me more insights in what went wrong. The AoT-compilation does a lot more ahead of time (of course).
I'm not sure this is correct but somewhere in the AoT-compilation errors it said that provides only could be of type Provide or Type, I registered a lot of services in my modules as just the service like this:

    // rest of the module declaration
    providers: [
        MyService,
        MyOtherService
    ]
    // rest of the module declaration
I changed all that with

    // rest of the module declaration
    providers: [
        { provide: MyService, useClass: MyService }
        { provide: MyOtherService, useClass: MyOtherService }
    ]
    // rest of the module declaration
I don't know if this is the correct way, but this gave me errors that came after the Provide-error. So I was one step further. I had some problems with my routing too and stuff. The AoT-compilation doesn't like it when using certain functions in the registration steps.
I hope the -aot option helps you. I can build now with or without AoT but with is better of course.
